application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,311 @@
+ * <p>
+ * To include a new RSS feed, simply define an {@link RSSFeed} entry in the {@code "rssFeeds"} array
+ * within the configuration file, adhering to the format shown below:
+ * 
+ * <pre>
+ * {@code
+ * {
+ *     "url": "https://example.com/feed",
+ *     "targetChannelPattern": "example",
+ *     "dateFormatterPattern": "EEE, dd MMM yyyy HH:mm:ss Z"
+ * }
+ * }
+ * </pre>
+ * 
+ * Where:
+ * <ul>
+ * <li>{@code url} represents the URL of the RSS feed.</li>
+ * <li>{@code targetChannelPattern} specifies the pattern to identify the target channel for the
+ * feed posts.</li>
+ * <li>{@code dateFormatterPattern} denotes the pattern for parsing the date and time information in
+ * the feed.</li>
+ * </ul>
+ */
+public final class RSSHandlerRoutine implements Routine {
+
+    private static final Logger logger = LoggerFactory.getLogger(RSSHandlerRoutine.class);
+    private static final RssReader RSS_READER = new RssReader();
+    private static final int MAX_CONTENTS = 1000;
+    private static final ZonedDateTime ZONED_TIME_MIN =
+            ZonedDateTime.of(LocalDateTime.MIN, ZoneId.systemDefault());
+    private final List<RSSFeed> feeds;
+    private final Predicate<String> defaultChannelPattern;
+    private final Map<RSSFeed, Predicate<String>> targetChannelPatterns = new HashMap<>();
+    private final int interval;
+    private final Database database;
+
+    /**
+     * Constructs an RSSHandlerRoutine with the provided configuration and database.
+     *
+     * @param config The configuration containing RSS feed details.
+     * @param database The database for storing RSS feed data.
+     */
+    public RSSHandlerRoutine(Config config, Database database) {
+        this.feeds = config.getRssFeeds();
+        this.interval = config.getRssPollInterval();
+        this.database = database;
+        this.defaultChannelPattern =
+                Pattern.compile(config.getJavaNewsChannelPattern()).asMatchPredicate();
+        this.feeds.forEach(feed -> {
+            if (feed.targetChannelPattern() != null) {
+                Predicate<String> predicate =
+                        Pattern.compile(feed.targetChannelPattern()).asMatchPredicate();
+                targetChannelPatterns.put(feed, predicate);
+            }
+        });
+    }
+
+    @Override
+    @NotNull
+    public Schedule createSchedule() {
+        return new Schedule(ScheduleMode.FIXED_DELAY, 0, interval, TimeUnit.MINUTES);
+    }
+
+    @Override
+    public void runRoutine(@Nonnull JDA jda) {
+        feeds.forEach(feed -> sendRSS(jda, feed));
+    }
+
+    /**
+     * Sends all the necessary posts from a given RSS feed.
+     * <p>
+     * This handles fetching the latest posts from the given URL, checking which ones have already
+     * been posted by reading information from the database and updating the last posted date.
+     */
+    private void sendRSS(JDA jda, RSSFeed feedConfig) throws DateTimeParseException {
```

Comment:
It's weird that this `throws` clause about DateTimeParseException only appears here and nowhere else. Instead, put it where this exception can be thrown, not just here.